src/renderer/components/dialog/publicationInfos/opdsControls.tsx

```diff
@@ -29,21 +29,52 @@ export class OpdsControls extends React.Component<IProps> {
             return <></>;
         }
 
-        return publication.isFree ? (
-            <button
-                onClick={() => verifyImport(publication as ImportOpdsPublication)}
-                className={styles.lire}
-            >
-                {__("catalog.addBookToLib")}
-            </button>
-        ) : publication.hasSample && (
-            <button
-                onClick={() => verifyImport(publication as ImportOpdsPublication, true)}
-                className={styles.lire}
-            >
-                {__("opds.menu.addExtract")}
-            </button>
-        );
+        return <>
+            {publication.isFree ? (
+                <button
+                    onClick={() => verifyImport(publication as ImportOpdsPublication)}
+                    className={styles.lire}
+                >
+                    {__("catalog.addBookToLib")}
+                </button>
+            ) : publication.hasSample && (
+                <button
+                    onClick={() => verifyImport(publication as ImportOpdsPublication, true)}
+                    className={styles.lire}
+                >
+                    {__("opds.menu.addExtract")}
+                </button>
+            )}
+            <ul className={styles.liens}>
```

Comment:
If `!publication.buyUrl && !publication.borrowUrl && !publication.subscribeUrl`, the `ul` will be empty. So I think there should be a conditional statement to prevent the `ul` from being rendered when it is empty.